### ISSUE TITLE
Guru Meditation Error ?

### ISSUE THREAD

**Issue by advadv00** [2dom/PxMatrix#244]
I don't know if I can ask a question here.

I want to play a gif file with this pxmatrix library.
I modified this example to pxmatrix.
Animated GIF Decoding Example : 
https://github.com/mrfaptastic/ESP32-RGB64x32MatrixPanel-I2S-DMA/tree/master/examples/AnimatedGIFPanel

But there's an error message like this on the serial monitor.

board : WEMOS D1 MINI ESP32 

Serial Monitor error code : 

/home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/queue.c:1446 (xQueueGenericReceive)- assert failed!
abort() was called at PC 0x40087fdb on core 1

Backtrace: 0x4008b744:0x3ffbe6a0 0x4008b971:0x3ffbe6c0 0x40087fdb:0x3ffbe6e0 0x400d375e:0x3ffbe720 0x400d2e85:0x3ffbe740 0x400d16b5:0x3ffbe760 0x400d18b1:0x3ffbe790 0x4008128d:0x3ffbe7b0 0x4008468d:0x3ffbe7d0 0x4000bfed:0x3ffb1bb0 0x40089185:0x3ffb1bc0 0x40089720:0x3ffb1be0 0x40086864:0x3ffb1c00 0x40086d62:0x3ffb1c20 0x400f16fa:0x3ffb1c90 0x400da1e9:0x3ffb1cc0 0x400da41f:0x3ffb1cf0 0x400db98b:0x3ffb1d20 0x400dd506:0x3ffb1d60 0x400d9dd4:0x3ffb1da0 0x400d8b1c:0x3ffb1dd0 0x400e1285:0x3ffb1e30 0x4000bcc5:0x3ffb1e50 0x400e25e1:0x3ffb1e70 0x400d2776:0x3ffb1e90 0x400d1f0b:0x3ffb1f10 0x400d1b47:0x3ffb1f40 0x400d4465:0x3ffb1fb0 0x400882b5:0x3ffb1fd0

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:8896
load:0x40080400,len:5816
entry 0x400806ac
Starting AnimatedGIFs Sketch
 * Loading SPIFFS
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed)
Core 1 register dump:
PC      : 0x400d189c  PS      : 0x00060034  A0      : 0x40084690  A1      : 0x3ffbe7b0  
A2      : 0x00000000  A3      : 0x00000001  A4      : 0x000000ff  A5      : 0x4008a658  
A6      : 0x3ff42000  A7      : 0x700000bb  A8      : 0x80081290  A9      : 0x3ff5f000  
A10     : 0x3ffbec24  A11     : 0x20000000  A12     : 0x00000400  A13     : 0x3ffb1b20  
A14     : 0x00000001  A15     : 0x3ffb8058  SAR     : 0x0000001f  EXCCAUSE: 0x00000007  
EXCVADDR: 0x00000000  LBEG    : 0x4000c28c  LEND    : 0x4000c296  LCOUNT  : 0x00000000  
Core 1 was running in ISR context:
EPC1    : 0x40062230  EPC2    : 0x00000000  EPC3    : 0x00000000  EPC4    : 0x400d189c

Backtrace: 0x400d189c:0x3ffbe7b0 0x4008468d:0x3ffbe7d0 0x4006222d:0x3ffb1b60 0x4008d74f:0x3ffb1b80 0x4008d786:0x3ffb1bb0 0x4008d846:0x3ffb1be0 0x4008e119:0x3ffb1c00 0x40086ec4:0x3ffb1c20 0x400f16fa:0x3ffb1c90 0x400da1e9:0x3ffb1cc0 0x400da41f:0x3ffb1cf0 0x400db98b:0x3ffb1d20 0x400dd506:0x3ffb1d60 0x400d9dd4:0x3ffb1da0 0x400d8b1c:0x3ffb1dd0 0x400e1285:0x3ffb1e30 0x4000bcc5:0x3ffb1e50 0x400e25e1:0x3ffb1e70 0x400d2776:0x3ffb1e90 0x400d1f0b:0x3ffb1f10 0x400d1b47:0x3ffb1f40 0x400d4465:0x3ffb1fb0 0x400882b5:0x3ffb1fd0

Rebooting...

My code:

#define FILESYSTEM SPIFFS
#include <SPIFFS.h>
#include <AnimatedGIF.h>
#include <PxMatrix.h>

#define MATRIX_WIDTH  64
#define MATRIX_HEIGHT  64

#ifdef ESP32

#define P_LAT 22
#define P_A 19
#define P_B 23
#define P_C 18
#define P_D 5
#define P_E 15
#define P_OE 16
hw_timer_t * timer = NULL;
portMUX_TYPE timerMux = portMUX_INITIALIZER_UNLOCKED;

#endif

#ifdef ESP8266

#include <Ticker.h>
Ticker display_ticker;
#define P_LAT 16
#define P_A 5
#define P_B 4
#define P_C 15
#define P_D 12
#define P_E 0
#define P_OE 2

#endif
// Pins for LED MATRIX

//PxMATRIX display(MATRIX_WIDTH,MATRIX_HEIGHT,P_LAT, P_OE,P_A,P_B,P_C);
//PxMATRIX display(6MATRIX_WIDTH,MATRIX_HEIGHT,P_LAT, P_OE,P_A,P_B,P_C,P_D);
PxMATRIX display(MATRIX_WIDTH,MATRIX_HEIGHT,P_LAT, P_OE,P_A,P_B,P_C,P_D,P_E);

// This defines the 'on' time of the display is us. The larger this number,
// the brighter the display. If too large the ESP will crash
uint8_t display_draw_time=10; //10-50 is usually fine

// Some standard colors
uint16_t myRED = display.color565(255, 0, 0);
uint16_t myGREEN = display.color565(0, 255, 0);
uint16_t myBLUE = display.color565(0, 0, 255);
uint16_t myWHITE = display.color565(255, 255, 255);
uint16_t myYELLOW = display.color565(255, 255, 0);
uint16_t myCYAN = display.color565(0, 255, 255);
uint16_t myMAGENTA = display.color565(255, 0, 255);
uint16_t myBLACK = display.color565(0, 0, 0);

uint16_t myCOLORS[8]={myRED,myGREEN,myBLUE,myWHITE,myYELLOW,myCYAN,myMAGENTA,myBLACK};

//#define INTERRUPT_ATTR IRAM_ATTR 

#ifdef ESP8266
// ISR for display refresh
void display_updater()
{
  display.displayTestPattern(display_draw_time);
  // display.displayTestPixel(display_draw_time);
  //display.display(display_draw_time);
}
#endif

#ifdef ESP32
void IRAM_ATTR display_updater(){
//void display_updater(){
  // Increment the counter and set the time of ISR
  portENTER_CRITICAL_ISR(&timerMux);
  display.display(display_draw_time);
  //display.displayTestPattern(display_draw_time);
  portEXIT_CRITICAL_ISR(&timerMux);
}
#endif

void display_update_enable(bool is_enable)
{

#ifdef ESP8266
  if (is_enable)
    display_ticker.attach(0.004, display_updater);
  else
    display_ticker.detach();
#endif

#ifdef ESP32
  if (is_enable)
  {
    timer = timerBegin(0, 80, true);
    timerAttachInterrupt(timer, &display_updater, true);
    timerAlarmWrite(timer, 4000, true);
    timerAlarmEnable(timer);
  }
  else
  {
    timerDetachInterrupt(timer);
    timerAlarmDisable(timer);
  }
#endif
}

AnimatedGIF gif;
File f;
int x_offset, y_offset;

// Draw a line of image directly on the LED Matrix
void GIFDraw(GIFDRAW *pDraw)
{
    uint8_t *s;
    uint16_t *d, *usPalette, usTemp[320];
    int x, y, iWidth;

  iWidth = pDraw->iWidth;
  if (iWidth > MATRIX_WIDTH)
      iWidth = MATRIX_WIDTH;

    usPalette = pDraw->pPalette;
    y = pDraw->iY + pDraw->y; // current line
    
    s = pDraw->pPixels;
    if (pDraw->ucDisposalMethod == 2) // restore to background color
    {
      for (x=0; x<iWidth; x++)
      {
        if (s[x] == pDraw->ucTransparent)
           s[x] = pDraw->ucBackground;
      }
      pDraw->ucHasTransparency = 0;
    }
    // Apply the new pixels to the main image
    if (pDraw->ucHasTransparency) // if transparency used
    {
      uint8_t *pEnd, c, ucTransparent = pDraw->ucTransparent;
      int x, iCount;
      pEnd = s + pDraw->iWidth;
      x = 0;
      iCount = 0; // count non-transparent pixels
      while(x < pDraw->iWidth)
      {
        c = ucTransparent-1;
        d = usTemp;
        while (c != ucTransparent && s < pEnd)
        {
          c = *s++;
          if (c == ucTransparent) // done, stop
          {
            s--; // back up to treat it like transparent
          }
          else // opaque
          {
             *d++ = usPalette[c];
             iCount++;
          }
        } // while looking for opaque pixels
        if (iCount) // any opaque pixels?
        {
          for(int xOffset = 0; xOffset < iCount; xOffset++ ){
            display.drawPixelRGB565(x + xOffset, y, usTemp[xOffset]);
          }
          x += iCount;
          iCount = 0;
        }
        // no, look for a run of transparent pixels
        c = ucTransparent;
        while (c == ucTransparent && s < pEnd)
        {
          c = *s++;
          if (c == ucTransparent)
             iCount++;
          else
             s--; 
        }
        if (iCount)
        {
          x += iCount; // skip these
          iCount = 0;
        }
      }
    }
    else // does not have transparency
    {
      s = pDraw->pPixels;
      // Translate the 8-bit pixels through the RGB565 palette (already byte reversed)
      for (x=0; x<pDraw->iWidth; x++)
      {
        display.drawPixelRGB565(x, y, usPalette[*s++]);
      }
    }
} /* GIFDraw() */

void * GIFOpenFile(const char *fname, int32_t *pSize)
{
  f = FILESYSTEM.open(fname);
  if (f)
  {
    *pSize = f.size();
    return (void *)&f;
  }
  return NULL;
} /* GIFOpenFile() */

void GIFCloseFile(void *pHandle)
{
  File *f = static_cast<File *>(pHandle);
  if (f != NULL)
     f->close();
} /* GIFCloseFile() */

int32_t GIFReadFile(GIFFILE *pFile, uint8_t *pBuf, int32_t iLen)
{
    int32_t iBytesRead;
    iBytesRead = iLen;
    File *f = static_cast<File *>(pFile->fHandle);
    // Note: If you read a file all the way to the last byte, seek() stops working
    if ((pFile->iSize - pFile->iPos) < iLen)
       iBytesRead = pFile->iSize - pFile->iPos - 1; // <-- ugly work-around
    if (iBytesRead <= 0)
       return 0;
    iBytesRead = (int32_t)f->read(pBuf, iBytesRead);
    pFile->iPos = f->position();
    return iBytesRead;
} /* GIFReadFile() */

int32_t GIFSeekFile(GIFFILE *pFile, int32_t iPosition)
{ 
  int i = micros();
  File *f = static_cast<File *>(pFile->fHandle);
  f->seek(iPosition);
  pFile->iPos = (int32_t)f->position();
  i = micros() - i;
//  Serial.printf("Seek time = %d us\n", i);
  return pFile->iPos;
} /* GIFSeekFile() */

unsigned long start_tick = 0;

void ShowGIF(char *name)
{
  start_tick = millis();
   
  if (gif.open(name, GIFOpenFile, GIFCloseFile, GIFReadFile, GIFSeekFile, GIFDraw))
  {
    x_offset = (MATRIX_WIDTH - gif.getCanvasWidth())/2;
    if (x_offset < 0) x_offset = 0;
    y_offset = (MATRIX_HEIGHT - gif.getCanvasHeight())/2;
    if (y_offset < 0) y_offset = 0;
    Serial.printf("Successfully opened GIF; Canvas size = %d x %d\n", gif.getCanvasWidth(), gif.getCanvasHeight());
    Serial.flush();
    while (gif.playFrame(true, NULL))
    {      
      if ( (millis() - start_tick) > 20000) { // we'll get bored after about 20 seconds of the same looping gif
        break;
      }
    }
    gif.close();
  }

} /* ShowGIF() */

void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  
  Serial.println("Starting AnimatedGIFs Sketch");

  // Start filesystem
  Serial.println(" * Loading SPIFFS");
  if(!SPIFFS.begin()){
        Serial.println("SPIFFS Mount Failed");
  }
  display_update_enable(true);
//  
//  #ifdef ESP8266
//    display_ticker.attach(0.004, display_updater);
//  #endif
//
//  #ifdef ESP32
//    timer = timerBegin(0, 80, true);
//    timerAttachInterrupt(timer, &display_updater, true);
//    timerAlarmWrite(timer, 4000, true);
//    timerAlarmEnable(timer);
//  #endif
//  
  display.begin(32);
  //display.flushDisplay();
  display.clearDisplay();
  display.setTextColor(myCYAN);
  display.setCursor(2,0);
  display.print("Pixel");
  display.setTextColor(myMAGENTA);
  display.setCursor(2,16);
  display.println("hello");

  delay(1000);

  gif.begin(LITTLE_ENDIAN_PIXELS);
  
}

void loop() {
char *szDir = "/gifs"; // play all GIFs in this directory on the SD card
char fname[256];
File root, temp;

   while (1) // run forever
   {
      root = FILESYSTEM.open(szDir);
//      if (root)
//      {
//         temp = root.openNextFile();
//            while (temp)
//            {
//              if (!temp.isDirectory()) // play it
//              {
//                strcpy(fname, temp.name());
//
//                Serial.printf("Playing %s\n", temp.name());
//                Serial.flush();
//                ShowGIF((char *)temp.name());
//             
//              }
//              temp.close();
//              temp = root.openNextFile();
//            }
//         root.close();
//      } // root
      delay(4000); // pause before restarting
   } // while
}

**Comment by tsondie21** [comment]
I was getting guru mediation error as well and I lowered my display_draw_time and that fixed it for me.
